### PR TITLE
Fix content length in CONNECT requests and responses

### PR DESCRIPTION
**Pull request by mmatczuk**
Make sure that we correctly read and send Content-Length header value for CONNECT requests.

### CODE REVIEW

**Comment by mmatczuk** [comment]
I need to add test here.

File: internal/martian/proxy_connect.go [hunk 1]
```diff
@@ -122,5 +121,19 @@ func (p *Proxy) connectSOCKS5(req *http.Request, proxyURL *url.URL) (*http.Respo
 		return nil, nil, err
 	}
 
-	return proxyutil.NewResponse(200, http.NoBody, req), conn, nil
+	return newConnectResponse(req), conn, nil
+}
+
+func newConnectResponse(req *http.Request) *http.Response {
+	return &http.Response{
+		StatusCode: http.StatusOK,
```
Comment (Choraden) [inline internal/martian/proxy_connect.go:129]: Please set `Status` too. The response log lacks it, you can see it in commit message.
```
sc-1         | HTTP/2.0
```
vs 
```
sc-server-1  | HTTP/1.1 200 OK
```

Comment (mmatczuk) [reply internal/martian/proxy_connect.go:143]: Done and donner

Comment (Choraden) [reply internal/martian/proxy_connect.go:143]: Wrong 🤔. That's how it's done in proxyutil 
```
Status:     fmt.Sprintf("%d %s", code, http.StatusText(code)),
```

**Comment by mmatczuk** [comment]
> I wonder if we still need that, since we set the ContentLength in read. I understand this is just a sanity check if the ContentLength was modified in Martian req/res modifiers?

I'd leave that to be on the safe side, we can add some logging.

File: internal/martian/proxy_connect.go (hunk 1 above)
Comment (mmatczuk) [reply internal/martian/proxy_connect.go]: That is the best way of doing it, fixed.

**Comment by Choraden** [comment]
Sure, let's log it.

**Review by Choraden — APPROVED** [review]
LGTM

**Pull request merged**